benchmark_v2/run_benchmarks.py

```diff
@@ -192,10 +198,111 @@ def generate_summary_report(
     return summary_file
 
 
+def upload_results_to_hf_dataset(
+    output_dir: str,
+    summary_file: str,
+    dataset_name: str,
+    run_id: Optional[str] = None,
+    logger: logging.Logger = None
+) -> Optional[str]:
+    """
+    Upload benchmark results to a HuggingFace Dataset.
+    Based on upload_collated_report() from utils/collated_reports.py
+    
+    Args:
+        output_dir: Local output directory containing results
+        summary_file: Path to the summary file
+        dataset_name: Name of the HuggingFace dataset to upload to
+        run_id: Unique run identifier (if None, will generate one)
+        logger: Logger instance
+        
+    Returns:
+        The run_id used for the upload, None if upload failed
+    """
+    if logger is None:
+        logger = logging.getLogger(__name__)
+    
+    import os
+    from huggingface_hub import HfApi
+    
+    api = HfApi()
+    
+    if run_id is None:
+        github_run_number = os.getenv("GITHUB_RUN_NUMBER")
+        github_run_id = os.getenv("GITHUB_RUN_ID")
+        if github_run_number and github_run_id:
+            run_id = f"{github_run_number}-{github_run_id}"
```

Comment:
```suggestion
        github_run_number = os.getenv("GITHUB_RUN_NUMBER")
        github_run_id = os.getenv("GITHUB_RUN_ID")
        if github_run_number and github_run_id:
            run_id = f"{github_run_number}-{github_run_id}"
        else:
            run_id = uuid.uuid4()
```
wdyt?

Reply:
or `str(uuid.uuid4())[:8]` as you used below

Reply:
The else case is handled here with the benchmark ID:
https://github.com/huggingface/transformers/pull/40716/files#diff-94fe5532c68b73efc8544ea11c0fa4d90a1d3e9266a8f8ccee2389752cea3d7cR475-R482